precompiles/ArbNativeToken.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021-2025, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+package precompiles
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/core/tracing"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+)
+
+// ArbNativeToken precompile enables minting and burning native tokens.
+// All calls to this precompile are authorized by the NativeTokenPrecompile wrapper.
+type ArbNativeToken struct {
+	Address addr
+}
+
+// Mints some amount of the native gas token for this chain to the given address
+func (con ArbNativeToken) MintNativeToken(c ctx, evm mech, to addr, amount huge) error {
+	if c.State.ArbOSVersion() < params.ArbosVersion_41 {
```

Comment:
I think we should use `PrecompileMethod.arbosVersion` instead of those checks here too (I added more details in ArbOwner.go comment).

Reply:
for `ArbNativeToken` we will probably want to use `Precompile.arbosVersion` instead

Reply:
Done, now only relying on authorization done in NativeTokenPrecompile) Call